tests/func/test_diff.py

```diff
@@ -163,17 +163,49 @@ def test_diff_no_cache(tmp_dir, scm, dvc):
         {"dir": {"file": "file content"}}, commit="first"
     )
     scm.tag("v1")
+
     tmp_dir.dvc_gen(
         {"dir": {"file": "modified file content"}}, commit="second"
     )
+    scm.tag("v2")
 
-    remove(first(stage.outs).cache_path)
-    remove("dir")
+    remove(dvc.cache.local.cache_dir)
 
     # invalidate_dir_info to force cache loading
     dvc.cache.local._dir_info = {}
 
-    diff = dvc.diff("v1")
+    diff = dvc.diff("v1", "v2")
```

Comment:
The change in this test is in order to actually test `diff` when there is no dir cache, and not intermix it with dirty repo `diff`.